modules/references/tests/server/reference-read-one.server.routes.tests.js

```diff
@@ -1,19 +1,234 @@
+'use strict';
+
+var _ = require('lodash'),
+    mongoose = require('mongoose'),
+    path = require('path'),
+    request = require('supertest'),
+    should = require('should'),
+    sinon = require('sinon'),
+    utils = require(path.resolve('./testutils/data.server.testutils')),
+    userProfile = require(path.resolve('./modules/users/server/controllers/users.profile.server.controller')),
+    express = require(path.resolve('./config/lib/express'));
+
 describe('Read a single reference by reference id', function () {
   // GET /references/:referenceId
   // logged in public user can read a single public reference by id
   // .....                 can read a single private reference if it is from self
   // logged in public user can not read other private references
+  var app = express.init(mongoose.connection);
+  var agent = request.agent(app);
+
+  var _usersPublic = utils.generateUsers(3, { public: true });
+  var _usersPrivate = utils.generateUsers(1, { public: false, username: 'private', email: 'non@example.com' });
+  var _users = _.concat(_usersPublic, _usersPrivate);
+
+  var users,
+      references;
+
+  beforeEach(function () {
+    sinon.useFakeTimers({ now: new Date('2019-01-13 13:21:55.1'), toFake: ['Date'] });
+  });
+
+  afterEach(function () {
+    sinon.restore();
+  });
+
+  beforeEach(function (done) {
+    utils.saveUsers(_users, function (err, usrs) {
+      users = usrs;
+      return done(err);
+    });
+  });
+
+  /**
+   *   0 1 2
+   * 0 . T F
+   * 1 F . T
+   * 2 T F .
```

Comment:
I get this is ids and such but the comment could be a little bit less cryptic. :-)

Reply:
Improved in a4f41ec